base/src/main/scala/data/Maybe.scala

```diff
@@ -4,11 +4,11 @@ package data
 sealed abstract class Maybe[A] {
   final def fold[B](f: A => B, b: => B): B = this match {
     case Maybe.Just(a)  => f(a)
-    case Maybe.Empty()  => b
+    case _              => b
   }
 }
 
 object Maybe extends MaybeFunctions with MaybeInstances with MaybeSyntax {
-  final case class Empty[A]() extends Maybe[A]
+  final private[data] case object Empty extends Maybe[Nothing]
```

Comment:
Lost in a force push, but @puffnfresh was asking about hiding `Just` as well here.

For which I say that I'm waiting on the result of investigation with optics, to then choose an appropriate strategy and review all datatype constructors.